### ISSUE TITLE
Docs improvement: Include scopes required to authenticate the action in the readme

### ISSUE THREAD

**Issue by bram-arcsec** [tailscale/github-action#110]
The README recommends the use of the OAuth API to register the client. There seems to be no fine grained permissions to limit the usage of the OAuth credentials to only register devices to a certain tag. If the key is exposed somehow it can impact the security of your Tailnet. 

With the authkey you can restrict the devices to a certain tag (that can be used in a ACL to restrict the runners to certain parts of the network).

What was the reasoning to retire the authkey functionality?

**Comment by evilhamsterman** [comment]
OAuth clients can restrict devices to certain tags, in fact you are are required to use tags unlike Auth keys which you can use without tags.

From the documentation https://tailscale.com/kb/1215/oauth-clients#generating-long-lived-auth-keys

> When you create an OAuth client with the scope devices, you must select one or more [ACL tags](https://tailscale.com/kb/1068/acl-tags/), which can be any tag or set of tags in your tailnet. Auth keys created with this client must have those exact tags, or tags owned by the client’s tags. Additionally, these tags need to be specified in the API call.

And when you generate an OAuth client with `device:write` permissions it forces you to select a tag(s) before it will generate the client 
![image](https://github.com/tailscale/github-action/assets/59560/9d4d8e39-242a-4628-866d-9e932cb5ae16)

**Comment by GustavoKatel** [comment]
this info should be on the readme

**Comment by sylr** [comment]
> this info should be on the readme

Absolutely, the README should explain what are the least scopes needed for an Oauth Client.

**Comment by samlinville** [comment]
Hi, I have renamed the issue to focus on the ask, which is better documentation in the README for setting up OAuth clients to work with the action. Thank you for raising this!
